src/utils/mapReservationData.ts

```diff
@@ -0,0 +1,46 @@
+import { Apartment, CustomerReservation, Project } from '../types';
+
+export const groupReservationsByProject = (reservations: CustomerReservation[]) => {
+  if (!reservations || !reservations.length) return [];
+
+  // Use project_uuid to group each reservation under one project
+  const reservationsByProjectUUID = (): any[] => {
+    const grouped = reservations.reduce((r: any, reservation: CustomerReservation) => {
+      r[reservation.project_uuid] = r[reservation.project_uuid] || [];
+      r[reservation.project_uuid].push(reservation);
+      return r;
+    }, {});
```

Comment:
Hmm, wouldn't something like `const grouped = reservations.group_by(({ project_uuid }) => project_uuid);` do the same?

Reply:
Discussed about this:

`Array.prototype.groupBy()` is not yet natively supported by all major browsers and we probably don't want to install lodash just for this.